### PR TITLE
task4-articles

### PR DESCRIPTION
**Pull request by Amir-Rsh**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,35 +1,28 @@
...
+import axios from "axios";
+import ListOfArticles from "./Components/ListOfArticles";
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [articles, setArticles] = useState([]);
+
+  useEffect(() => {
+    axios
+      .get("https://nc-news-by-amir.onrender.com/api/articles")
```
Comment (itskatherine) [inline src/App.jsx:12]: Great use of axios, you should separate out your API calls into functions in it's own `api.js` file, then import them where needed around your app.
This way you can just import axios into one file as well as separating out your concerns 👍 
Should this API call happen here or in the `ListOfArticles`? 🤔

File: src/App.jsx [hunk 2]
```diff
@@ -1,35 +1,28 @@
...
-import './App.css'
+import { useEffect, useState } from "react";
+import Nav from "./Components/Nav";
+import { Routes, Route } from "react-router-dom";
+import axios from "axios";
+import ListOfArticles from "./Components/ListOfArticles";
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [articles, setArticles] = useState([]);
```
Comment (itskatherine) [inline src/App.jsx:8]: Does this state need to be in the App component, or could it be initialised in `ListOfArticles`?

File: src/Components/ListOfArticles.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import ArticleCard from "./ArticleCard";
+import HomeHeader from "./HomeHeader";
+
+export default function ListOfArticles({ articles }) {
```
Comment (itskatherine) [inline src/Components/ListOfArticles.jsx:4]: Can you please use an `isLoading` state to conditionally render something to make sure the user knows the articles are loading?

**Review by itskatherine — COMMENTED** [review]
Solid styling thoughout Amir! Love your use of bootstrap and site is behaving responsively so far 📱 
Please have a look at my comments though, it pays to think a bit more carefully about where state needs to be. Also please add loading states/messages here and throughout your app going forward where necessary 👍

File: src/App.jsx (hunk 2 above)
Comment (Amir-Rsh) [reply src/App.jsx:8]: thank you for your feedback Katherine. sorry just a quick question. I have the whole list of articles showing on the homepage but I was also planning to add a 'my articles' page in the Nav that would need access to articles to show the user all of their own articles. would that be a good reason to put the state in the app?

**Pull request merged**